### ISSUE TITLE
Change or add option to use Map Elevation 

### ISSUE THREAD

**Issue by rouroni1** [cgeo/cgeo#15566]
### Tell us your idea!

I usualy hike when doing geocaching, and use offline Maps. And the Elevation that C:Geo uses is based in the GNSS data, but the diference is to big to map Elevation

If I'm at 100m and the summit is at 450m in the map I prefer to have the map elevation to know that I need to go up 350 m.

![Screenshot_2024-04-07-00-16-28-242_cgeo geocaching](https://github.com/cgeo/cgeo/assets/27059385/710fffee-d3f1-4a0b-8663-e1252e4ebee6)
.

### What solution would you suggest?

If it's possible add the option to use Map Elevation.

In the example picture:
Map Elevation: 12,5 m
GNSS Elevation: 70,0 m

**Comment by moving-bits** [comment]
The might be a slight misunderstanding with regard to what the meaning of "map elevation" is: It's not about "difference in elevation compared to current position", but "(absolute) elevation gathered by lidar data" in contrast to GPS elevation being "(absolute) elevation relativ to WGS84 elipsoid".

When we introduced the elevation display, we started with "map elevation" even for current position, but that has two drawbacks:
- "map elevation" is gathered from underlying map data and is a costly operation (which lead to significant delays in map display for some), whereas "GPS elevation" comes (more or less) "for free"
- "map elevation" and "GPS elevation" have different values in many places. On c:geo mainscreen we display "GPS elevation" for current position, so it might be confusing if we displayed "map elevation" for current position on the map.

Therefore the team decided to switch to "GPS elevation" for current position, and we adapted c:geo with the latest update.

Having said that, there is a different feature which might be for you:

With the latest update you can get an elevation chart for your individual route, which gives you a nice overview about elevation changes on your route. For this you need to
- use UnifiedMap (it's only implemented in that map type)
- create an individual route
- tap on the individual route

For a visual impression have a look at the screenshots in https://github.com/cgeo/cgeo/pull/14779#issuecomment-1793385182

**Comment by rouroni1** [comment]
Hi.

I've already tested the elevation chart, and for me works when I plan one route with c:geo, adding geocaches.

But, when I use a box track recorded with my watch (amazfit GTS), the elevation chart is empty.

And when hiking in mountain, it's hard to create one track adding geocaches, because most off the times its draws a straight line, messing up completely the chart, because straight lines dosen't count on the chart.


Here's some examples.

Empty chart with pre-recorded gpx track:
https://github.com/cgeo/cgeo/assets/27059385/06c527ce-718f-4095-9e22-86f8be0fccb4

Mountain:
https://github.com/cgeo/cgeo/assets/27059385/61780a60-5333-49e0-95f6-6bfd0d9655c2

City:
https://github.com/cgeo/cgeo/assets/27059385/114380f9-2afc-4def-af5f-82bb35d3f5b4

**Comment by moving-bits** [comment]
As of today, you can calculate the difference manually, by long-tapping the destination point, reading the "map elevation" value, then long-tapping current position, reading the "map elevation" value, and subtracting the two values.

But from what I read from your examples, I guess you want to have this automated:
- if this feature is activated (eg: "show map elevation difference to target"), and
- if a target is set (red line to destination)
- and map supports elevation data (not all do)
- then display the differences in map elevation between set target and current position below the blue position marker

Does this describe your feature request correctly?

**Comment by rouroni1** [comment]
Yes, that option also replaces the need to doo the mental calculation.

And should support negative values. If I'm at 100 m and the destination is at 90 m, then the value would be -10.0 m

**Comment by wolverine007** [comment]
@rouroni1 totally agree with you. The switch from "map elevation" to "GPS elevation" confused me as well.

at our home town we have something between 70-80m in average. 

Basing on the map elevation this hight was correctly displayed everywhere. 

Now with the GPS elevation being set as the new default, the hight is everywhere completly wrong - in average +50m higher than it should be.

So I also would like to switch back to the map elevation based hight.

**Comment by moving-bits** [comment]
@rouroni1 

> But, when I use a box track recorded with my watch (amazfit GTS), the elevation chart is empty.

can you cross-check if the GPX track file generated by your watch does contain elevation information? (`<ele>` tags for each track point).
- If no, then we don't have the data to display an elevation chart.
- If yes, then you got hit by a then-missing feature, as described in #15683, as c:geo did not read elevation info from GPX tracks. This has been fixed meanwhile (#15691), is available in our nightly versions and will be part of the next feature release (not bugfix release, though).

**Comment by moving-bits** [comment]
With regard to feature request proposed in https://github.com/cgeo/cgeo/issues/15566#issuecomment-2041404845, c:geo release 2024.04.25 has a new feature (see #15583):

Tap on a free space on the map ("destination") and select "selected position" in the popup that opens. You'll then get a new popup with some info about the position you have selected:

![image](https://github.com/cgeo/cgeo/assets/3754370/0fc14470-de07-449b-ae17-2d901d4af7c0)

- coordinates (tap on it for different formats)
- straight distance from current position
- elevation difference to current position <= that's what you are looking for
- map elevation

Although it's not selected target-specific, will this still suffice your needs?

**Comment by rouroni1** [comment]
> @rouroni1
> 
> > But, when I use a box track recorded with my watch (amazfit GTS), the elevation chart is empty.
> 
> can you cross-check if the GPX track file generated by your watch does contain elevation information? (`<ele>` tags for each track point).
> 
> * If no, then we don't have the data to display an elevation chart.
> * If yes, then you got hit by a then-missing feature, as described in [Elevation chart for tracks exported from Garmin Basecamp does not get drawn #15683](https://github.com/cgeo/cgeo/issues/15683), as c:geo did not read elevation info from GPX tracks. This has been fixed meanwhile ([Read elevation info from track (fix #15683) #15691](https://github.com/cgeo/cgeo/pull/15691)), is available in our nightly versions and will be part of the next feature release (not bugfix release, though).

@moving-bits 
I've checked. The GPX have the "<ele>" tag.

I'll wait for the next beta or release to test. :)
![elevation_gpx](https://github.com/cgeo/cgeo/assets/27059385/04b11516-e842-452c-8909-22b412819c0d)

**Comment by rouroni1** [comment]
> With regard to feature request proposed in [#15566 (comment)](https://github.com/cgeo/cgeo/issues/15566#issuecomment-2041404845), c:geo release 2024.04.25 has a new feature (see #15583):
> 
> Tap on a free space on the map ("destination") and select "selected position" in the popup that opens. You'll then get a new popup with some info about the position you have selected:
> 
> ![image](https://private-user-images.githubusercontent.com/3754370/331896280-0fc14470-de07-449b-ae17-2d901d4af7c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYyMTY2MjIsIm5iZiI6MTcxNjIxNjMyMiwicGF0aCI6Ii8zNzU0MzcwLzMzMTg5NjI4MC0wZmMxNDQ3MC1kZTA3LTQ0OWItYWUxNy0yZDkwMWQ0YWY3YzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MjBUMTQ0NTIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDU2ZDkyZDY1ZWMyMzBjOTI5OWEzZjU1MjU3MzVmYzFjOTBlMTk3NDQ2M2MyN2EwN2UzYjYxYzE3YjEzNjM2YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qLT90Cv4a_HYY9x5h2uwTUDO8M9NY05jhH3-9tjQHoQ)
> 
> * coordinates (tap on it for different formats)
> * straight distance from current position
> * elevation difference to current position <= that's what you are looking for
> * map elevation
> 
> Although it's not selected target-specific, will this still suffice your needs?

@moving-bits 
This will do. It involves looking at the map to choose your destination (scrolling the map), but it's already a big help.

**Comment by moving-bits** [comment]
Can we close this issue or is there anything left to do?

**Comment by rouroni1** [comment]
Along the implementations, if I recall correctly, something about android 14 and how it made possible to retrieve the right altitude.
I think it was this: https://github.com/cgeo/cgeo/pull/15898

So the "issue" is fixed. Thank you.

![Image](https://github.com/user-attachments/assets/409158bb-4076-4b1d-a2b0-5fd8dcb57500)
